pkg/api/v20170701/types.go

```diff
@@ -139,9 +139,18 @@ func (m *MasterProfile) UnmarshalJSON(b []byte) error {
 		m.Count = 1
 	}
 
-	// TODO:
-	// Need to set default values for
-	// OSDiskSizeGB, FirstConsecutiveStaticIP, StorageProfile
+	if m.FirstConsecutiveStaticIP == "" {
+		// if FirstConsecutiveStaticIP is missing, set to default 10.240.255.5
+		m.FirstConsecutiveStaticIP = "10.240.255.5"
+	}
+
+	if m.StorageProfile == "" {
+		// if StorageProfile is missing, set to default StorageAccount
+		m.StorageProfile = StorageAccount
+	}
+
+	// OSDiskSizeGB
+	// The current behavior is, when OSDiskSizeGB is 0, we just don't pass it in the arm template
```

Comment:
@anhowe @amanohar ,
Wondering what the default value would be used for OSDiskSizeGB? OSDiskSizeGB is different with other fields, such as "StorageProfile". I can deduce StorageProfile from either template, or SetDefaultProperties function. However, OSDiskSizeGB will be strip out in the arm template if it is 0.

Options:
1. We don't have a default value defined in the api spec. (I don't like this. User would be confused)
2. We pick a value as default. (What the value would be 200GB?)